azurerm/resource_arm_eventhub_namespace.go

```diff
@@ -283,3 +366,96 @@ func eventHubNamespaceStateStatusCodeRefreshFunc(ctx context.Context, client *ev
 		return res, strconv.Itoa(res.StatusCode), nil
 	}
 }
+
+func expandEventHubNamespaceNetworkRuleset(input []interface{}) *eventhub.NetworkRuleSetProperties {
+	if len(input) == 0 {
+		return nil
+	}
+
+	block := input[0].(map[string]interface{})
+
+	ruleset := eventhub.NetworkRuleSetProperties{
+		DefaultAction: eventhub.DefaultAction(block["default_action"].(string)),
+	}
+
+	if v, ok := block["virtual_network_rule"].([]interface{}); ok {
+		if len(v) > 0 {
+
+			var rules []eventhub.NWRuleSetVirtualNetworkRules
+			for _, r := range v {
+				rblock := r.(map[string]interface{})
+				rules = append(rules, eventhub.NWRuleSetVirtualNetworkRules{
+					Subnet: &eventhub.Subnet{
+						ID: utils.String(rblock["subnet_id"].(string)),
+					},
+					IgnoreMissingVnetServiceEndpoint: utils.Bool(rblock["ignore_missing_virtual_network_service_endpoint"].(bool)),
+				})
+			}
+
+			ruleset.VirtualNetworkRules = &rules
+		}
+	}
+
+	if v, ok := block["ip_rule"].([]interface{}); ok {
+		if len(v) > 0 {
+
+			var rules []eventhub.NWRuleSetIPRules
+			for _, r := range v {
+				rblock := r.(map[string]interface{})
+				rules = append(rules, eventhub.NWRuleSetIPRules{
+					IPMask: utils.String(rblock["ip_mask"].(string)),
+					Action: eventhub.NetworkRuleIPAction(rblock["action"].(string)),
+				})
+			}
+
+			ruleset.IPRules = &rules
+		}
+	}
+
```

Comment:
Do we want to do extra validation to ensure that one of `ip_rule` or `virtual_network_rule` are specified?

Reply:
It works just fine with neither 🙂